runtime/src/main/java/org/corfudb/util/MetricsUtils.java

```diff
@@ -199,17 +239,38 @@ public static void addCacheGauges(@NonNull MetricRegistry metrics,
         }
     }
 
-    public static void addJvmMetrics(@NonNull MetricRegistry metrics, String pfx) {
+    // If enabled, setup reporting of JVM metrics including garbage collection,
+    // memory, and thread statistics.
+    private static void setupJvmMetrics(@NonNull MetricRegistry metrics) {
+        if (!metricsJvmCollectionEnabled) return;
+
         try {
-            metrics.register(pfx + "jvm.gc", metricsJVMGC);
-            metrics.register(pfx + "jvm.memory", metricsJVMMem);
-            metrics.register(pfx + "jvm.thread", metricsJVMThread);
-            metrics.register(pfx + "jvm.file-descriptors-used", metricsJVMFdGauge);
+            metrics.register("jvm.gc", metricsJVMGC);
```

Comment:
I think we should only collect/report corfu specific metrics for the client and that the consuming application should take care of these jvm metrics. For example, enable this for corfu, would also report numbers from the application (since they use the same jvm).

Reply:
I have the same question. please let us know.

Reply:
Currently NSX doesn’t provide such metrics and having this metrics provides a useful insight and a troubleshooting tool into monitoring of the memory and threading metrics. Also the collection/reporting of jvm metrics are disabled by default and can be configured flexibly and independently according to the needs for such metrics on clients or servers.